src/actions/Scroll.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  Action,
+  ActionPreferences,
+  ActionReturnValue,
+  Graph,
+  TypedSelection,
+  SelectionWithEditor,
+} from "../Types";
+import { displaySelectionDecorations } from "../editDisplayUtils";
+import { groupBy } from "../itertools";
+import { commands, TextEditor, Selection, window } from "vscode";
+import { focusEditor } from "./setSelectionsAndFocusEditor";
+
+class Scroll implements Action {
+  targetPreferences: ActionPreferences[] = [{ insideOutsideType: "inside" }];
+
+  constructor(private graph: Graph, private at: string) {
+    this.run = this.run.bind(this);
+  }
+
+  async run([targets]: [TypedSelection[]]): Promise<ActionReturnValue> {
+    const selectionGroups = groupBy(
+      targets,
+      (t: TypedSelection) => t.selection.editor
+    );
+
+    const selections = <SelectionWithEditor[]>[];
+    const lines = <LineWithEditor[]>[];
+
+    [...selectionGroups].forEach(([editor, targets]) => {
+      const { lineNumber, lineNumberDecorate } = getLineNumber(
+        targets,
+        this.at
+      );
+      lines.push({ lineNumber, editor });
+      selections.push({
+        selection: new Selection(lineNumberDecorate, 0, lineNumberDecorate, 0),
+        editor,
+      });
+    });
+
+    await displaySelectionDecorations(
+      selections,
+      this.graph.editStyles.referencedLine
+    );
+
+    const originalEditor = <TextEditor>window.activeTextEditor;
```

Comment:
why the cast?

Reply:
Just to make sure it doesn't complain about possible undefined.

Reply:
But it might actually be undefined right?  Where does it complain about possibly undefined?

Reply:
Unlikely but I added a check for it instead of the cast. Is used below when focusing back the original editor.